### PR TITLE
Allow restore from web user to mobile user for edit forms

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke (mind taking a look at this while cory is gone?) @wpride this should enable a web user to restore as a mobile user as long as they have data cleanup permission or are superusers. this is necessary for the Edit Forms workflow with the new sqlite backend. essentially when we edit a form, touchforms sends a request to restore as the mobile user, but has the auth of the web worker, which makes sense. this allows that to happen without receiving an unauthorized error. 🎣 or 🏠 

http://manage.dimagi.com/default.asp?230455

cc: @proteusvacuum @sravfeyn

### CODE REVIEW

File: corehq/apps/ota/utils.py [hunk 1]
```diff
@@ -80,3 +82,46 @@ def demo_restore_date_created(commcare_user):
         restore = DemoUserRestore.objects.get(id=commcare_user.demo_restore_id)
         if restore:
             return restore.timestamp_created
+
+
+def is_permitted_to_restore(domain, couch_user, as_user, has_data_cleanup_privelege):
```
Comment (benrudolph) [inline corehq/apps/ota/utils.py]: this is a mess, but tests should clear up all the different code paths

File: corehq/apps/ota/views.py [hunk 2]
```diff
@@ -109,31 +110,35 @@ def get_restore_params(request):
         'version': request.GET.get('version', "1.0"),
         'state': request.GET.get('state'),
         'items': request.GET.get('items') == 'true',
-        'force_restore_mode': request.GET.get('mode')
+        'force_restore_mode': request.GET.get('mode'),
+        'as_user': request.GET.get('as'),
```
Comment (benrudolph) [inline corehq/apps/ota/views.py]: as user from cloudcare comes in as <user>@<domain> without the `commcarehq.org`

**Comment by benrudolph** [comment]
(note last commit has ci skip)

File: corehq/apps/ota/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            True,
+        )
+        self.assertFalse(is_permitted)
+        self.assertIsNotNone(message)
+
+    def test_web_user_as_user_bad_domain(self):
+        is_permitted, message = is_permitted_to_restore(
+            self.domain,
+            self.web_user,
+            u'{}@wrong-domain'.format(self.commcare_user),  # Malformed, should include domain
```
Comment (snopoke) [inline corehq/apps/ota/tests/test_utils.py]: line comment incorrect

File: corehq/apps/ota/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            False,
+        )
+        self.assertFalse(is_permitted)
+        self.assertIsNotNone(message)
+
+    def test_web_user_as_user_bad_username(self):
+        is_permitted, message = is_permitted_to_restore(
+            self.domain,
+            self.web_user,
+            u'{}'.format(self.commcare_user),  # Malformed, should include domain
```
Comment (snopoke) [inline corehq/apps/ota/tests/test_utils.py]: nit: don't need to use `format`

**Comment by snopoke** [comment]
nice

**Pull request merged**